sklearn/linear_model/_ridge.py

```diff
@@ -262,7 +261,7 @@ def ridge_regression(X, y, alpha, sample_weight=None, solver='auto',
         assumed to be specific to the targets. Hence they must correspond in
         number.
 
-    sample_weight : float or numpy array of shape (n_samples,), default=None
+    sample_weight : float or array-like of shape (n_samples,), default=None
```

Comment:
Passing float seems like a bug

Reply:
It is treated as `float * np.ones_like(y)`. So not really a bug. Any further thoughts?

Reply:
Yes, that's how we treat it in `_check_sample_weight`. Not sure there is a use-case for it, if not we could deprecate, but that's an issue orthogonal to this PR